lib/flagsmith/engine/features/models.rb

```diff
@@ -0,0 +1,172 @@
+# frozen_string_literal: true
+
+require_relative '../utils/hash_func'
+
+module Flagsmith
+  module Engine
+    # FeatureModel
+    class Feature
+      attr_reader :id, :name, :type
+
+      def initialize(id:, name:, type:)
+        @id = id
+        @name = name
+        @type = type
+      end
+
+      def ==(other)
+        return false if other.nil?
+
+        id == other.id
+      end
+
+      class << self
+        def build(json)
+          new(**json.slice(:id, :name, :type))
+        end
+      end
+    end
+
+    module Features
+      # MultivariateFeatureOptionModel
+      class MultivariateOption
+        attr_reader :value, :id
+
+        def initialize(value:, id: nil)
+          @value = value
+          @id = id
+        end
+
+        class << self
+          def build(json)
+            new(**json.slice(:id, :value))
+          end
+        end
+      end
+
+      # MultivariateFeatureStateValueModel
+      class MultivariateStateValue
+        attr_reader :id, :multivariate_feature_option, :percentage_allocation, :mv_fs_value_uuid
+
+        def initialize(id:, multivariate_feature_option:, percentage_allocation:, mv_fs_value_uuid: SecureRandom.uuid)
+          @id = id
+          @percentage_allocation = percentage_allocation
+          @multivariate_feature_option = multivariate_feature_option
+          @mv_fs_value_uuid = mv_fs_value_uuid
+        end
+
+        def <=>(other)
```

Comment:
I wasn't sure what this method's purpose was so I quickly looked it up. Seems like it should return -1, 0, 1 or nil but it doesn't seem to be what's happening here? 

I'm also not sure why it's necessary? We don't seem to define any comparison methods on the `MultivariateFeatureStateValueModel` in the python client.

Reply:
Here we use sorting. I saw this logic in node-js sdk but not sure if we need it
https://github.com/Flagsmith/flagsmith-ruby-client/blob/release/3.0.0/lib/flagsmith/engine/features/models.rb#L120

Reply:
Ah, ok, yes, I think we probably do need to be able to sort these objects 👍